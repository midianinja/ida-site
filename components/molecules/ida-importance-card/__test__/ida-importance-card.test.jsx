import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import IDAImportanceCard from '../ida-importance-card';

describe('IDAImportanceCard', () => {
  afterEach(cleanup);
  
  it('renders without crashing', async () => {
    const { getByTestId } = render(
      <IDAImportanceCard
        title="Test"
        iconSrc="/static/icon/ilustra-app-switch.svg"
        explanation="Test explanation about IDA"
      />
    );
    expect(getByTestId('icon-src')).toHaveAttribute('src');
  });

  it('matches snapshot card', () => { 
    const tree = renderer.create(
      <IDAImportanceCard
        title="Test"
        iconSrc="/static/icon/ilustra-app-switch.svg"
        explanation="Test explanation about IDA"
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

