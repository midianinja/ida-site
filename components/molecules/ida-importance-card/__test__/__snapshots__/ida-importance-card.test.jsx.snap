// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IDAImportanceCard matches snapshot card 1`] = `
<div
  className="card"
>
  <img
    alt=""
    className="icon"
    data-testid="icon-src"
    src="/static/icon/ilustra-app-switch.svg"
  />
  <h3
    className="title"
  >
    Test
  </h3>
  <p
    className="explanation"
  >
    Test explanation about IDA
  </p>
</div>
`;
